apps/dashboard/app/helpers/dashboard_helper.rb

```diff
@@ -19,4 +19,49 @@ def logo_image_tag(url)
   def invalid_clusters
     @invalid_clusters ||= OodCore::Clusters.new(OodAppkit.clusters.select { |c| not c.valid? })
   end
+
+  def xdmod?
+    Configuration.xdmod_integration_enabled?
+  end
+
+  def pinned_apps?
+    @pinned_apps.present?
+  end
+
+  def motd?
+    @motd.present?
+  end
+
+  def landing_page_layout
+    #FIXME: should sanitize the landing_page_layout or cast somethings to Array in the upper layers
+    Configuration.landing_page_layout || default_landing_page_layout
+  end
+
+  def render_widget(widget)
+    begin
+      render partial: widget.to_s
+    rescue SyntaxError, StandardError => e
+      render partial: "widget_error", locals: { error: e, widget: widget.to_s }
+    end
+  end
+
+  private
+
+  def default_landing_page_layout
+    if xdmod?
+      if pinned_apps? || motd?
+        left_column = { width: 8, widgets: ['pinned_apps', 'motd'] }
+        right_column = { width: 4, widgets: ['xdmod_widget_job_efficiency', 'xdmod_widget_jobs'] }
+      else
+        left_column = { width: 4, widgets: ['xdmod_widget_job_efficiency'] }
+        right_column = { width: 8, widgets: ['xdmod_widget_jobs'] }
+      end
+    else
```

Comment:
This would also support the default for app sharing where the MOTD is on the right:

```
elsif pinned_apps? && motd?
  left_column = { width: 8, widgets: ['pinned_apps'] }
  right_column = { width: 4, widgets: ['motd'] }
else
  left_column = { width: 12, widgets: ['pinned_apps', 'motd'] }
end
```

Reply:
Done.